cpp-client/deephaven/client/src/server/server.cc

```diff
@@ -467,15 +513,6 @@ void Server::releaseAsync(Ticket ticket, std::shared_ptr<SFCallback<ReleaseRespo
   sendRpc(req, std::move(callback), sessionStub(), &SessionService::Stub::AsyncRelease);
 }
 
-std::pair<std::string, std::string> Server::getAuthHeader() const {
-  return std::make_pair(authorizationKey, sessionToken_);
-}
-
-void Server::addSessionToken(grpc::ClientContext *ctx) {
-  std::lock_guard guard(mutex_);
-  ctx->AddMetadata(authorizationKey, sessionToken_);
-}
-
```

Comment:
Somewhere in this file please add
```
  void Server::forEachHeaderNameAndValue(const std::function<void(const std::string &, const std::string &)> &fun) {
    {
      std::lock_guard guard(mutex_);
      fun(authorizationKey, sessionToken_);
    }
    for (const auto &header : extraHeaders_) {
      fun(header.first, header.second);
    }
  }
```

By the way ... it's just a bit of an anti-pattern in my coding style to invoke callbacks under lock. Because something horrible can happen in the callback and then you have a deadlock. You might want to consider this more defensive approach

```
mutex_.lock();
auto tokenCopy = sessionToken_;
mutex_.unlock();
fun(authorizationKey, tokenCopy);
```

Reply:
Thanks, didn't think about that, and I consider that an anti pattern myself as well.